src/renderer/atlas/AtlasEngine.r.cpp

```diff
@@ -944,6 +924,8 @@ void AtlasEngine::_d2dDrawDirtyArea()
 
         // Draw background.
         {
+            _r.d2dRenderTarget->SetPrimitiveBlend(D2D1_PRIMITIVE_BLEND_COPY);

```

Comment:
2. `d2dMode` failed to work with transparent backgrounds

(D2D defaults to source-over alpha blending so the `_d2dFillRectangle` to fill the background wouldn't work if the background color is transparent: It would just layer colors on top of the existing content instead of replacing the existing content.)